source/SpinalHDL/Libraries/stream.rst

```diff
@@ -74,6 +74,16 @@ There is some examples of usage in SpinalHDL :
 * An UART controller could directly use the write port to drive UART pins and only consume the transaction at the end of the transmission.
   Be careful with that.
 
+Semantics
+---------
+
+When manually reading/driving the signals of a Stream keep in mind that:
+
+* After being asserted, ``valid`` may only be deasserted once the current payload was acknowleged. This means ``valid`` can only toggle to 0 the cycle after a the slave did a read by asserting ``ready``.
+* In contrast to that ``ready`` may change at any time. 
+* A transfer is only done on cycles where both ``valid`` and ``ready`` are asserted.
+* ``valid`` of a Stream must not combinatorically depend on ``ready`` and must be registered.
```

Comment:
You are right, it would actually be combinatorially ... which does not make it much better though.
After thinking about it: since ready may anyhow change it any time it does kinda not make sense to depend on it at all... Therefore I added that as a recommendation.